warehouse/ingest-core/src/test/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelperTest.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    /*
+     * Test that whindex fields and their values are correctly identified by and added to the WhindexFieldIngestHelper.
+     */
+    @Test
+    void testWhindexFieldDefinitionsParsing() {
+        Configuration config = new Configuration();
+
+        // delete src fields = true
+        config.set("wiki.whindex.rules.1.value_field", "AppleVF");
+        config.set("wiki.whindex.rules.1.src_field", "ForkSRC");
+        config.set("wiki.whindex.rules.1.delete_src_field", "true");
+        config.set("wiki.whindex.rules.1.dst_field", "ProvoloneDST");
+        config.set("wiki.whindex.rules.1.values", "kittyV,catV,satV,batV");
+        WhindexConfig expectedConfigA = new WhindexConfig();
+        expectedConfigA.setValueField("AppleVF");
+        expectedConfigA.setSourceField("ForkSRC");
+        expectedConfigA.setOverloaded(true);
+        expectedConfigA.setDestField("ProvoloneDST");
+        expectedConfigA.setValues(Arrays.asList("kittyV", "catV", "satV", "batV"));
+
+        config.set("wiki.whindex.rules.2.value_field", "aeiouVF");
+        config.set("wiki.whindex.rules.2.src_field", "qwertySRC");
+        config.set("wiki.whindex.rules.2.delete_src_field", "false");
+        config.set("wiki.whindex.rules.2.dst_field", "poiuyDST");
+        config.set("wiki.whindex.rules.2.values", "aV,bV,cV,dV");
+
+        WhindexConfig expectedConfigB = new WhindexConfig();
+        expectedConfigB.setValueField("aeiouVF");
+        expectedConfigB.setSourceField("qwertySRC");
+        expectedConfigB.setOverloaded(false);
+        expectedConfigB.setDestField("poiuyDST");
+        expectedConfigB.setValues(Arrays.asList("aV", "bV", "cV", "dV"));
+
+        config.set("wiki.whindex.rules.sillyrule.value_field", "hahaVF");
+        config.set("wiki.whindex.rules.sillyrule.src_field", "heheSRC");
+        config.set("wiki.whindex.rules.sillyrule.dst_field", "hohohoDST");
+        config.set("wiki.whindex.rules.sillyrule.values", "lolV,lololV,lolololV,lololololV");
+
+        WhindexConfig expectedConfigC = new WhindexConfig();
+        expectedConfigC.setValueField("hahaVF");
+        expectedConfigC.setSourceField("heheSRC");
+        expectedConfigC.setOverloaded(false);
+        expectedConfigC.setDestField("hohohoDST");
+        expectedConfigC.setValues(Arrays.asList("lolV", "lololV", "lolololV", "lololololV"));
+
+        Type type = new Type("wiki", null, null, null, 0, null);
+        WhindexFieldIngestHelper wHelper = new WhindexFieldIngestHelper(type);
+        wHelper.setup(config);
+
+        HashMultimap<String,WhindexConfig> expectedValues = HashMultimap.create();
+        expectedValues.put("AppleVF", expectedConfigA);
+        expectedValues.put("aeiouVF", expectedConfigB);
+        expectedValues.put("hahaVF", expectedConfigC);
+
+        Assertions.assertEquals(expectedValues, wHelper.getValueFieldsToWhindexConfigs());
+    }
+
+     @Test
+     void testGetWhindexFields() {
```

Comment:
This test contains assertions for three different scenarios. I recommend breaking up this test into three separate test cases, one for each scenario. It makes it clearer and easier to document what scenarios are being tested. Please do the same for the other tests in this class that do a similar thing.